docs-ref-services/key-vault.md

```diff
@@ -142,18 +111,140 @@ operation = client.vaults.create_or_update(
     }
 )
 
+# Run the operation
 vault = operation.result()
+
+# Show the URL of the created vault
 print(f'New vault URI: {vault.properties.vault_uri}')
 ```
 
 > [!div class="nextstepaction"]
 > [Explore the Management APIs](/python/api/overview/azure/keyvault/management)
 
-## Samples
-* [Manage Azure Key Vaults][1] 
-* [Azure Key Vault recovery][2]
+## Client library
+
+Install the library with `pip install azure-keyvault`.
+
+To access a key vault, an application must first authenticate with the Key Vault service. Authentication is best done using [Azure Managed Identities](/azure/active-directory/managed-identities-azure-resources/overview) (part of Azure Active Directory), and can also also be done with a service principal.
+
+To use Managed Identities, acquire a `DefaultAzureCredential` object. (To run this code on a developer workstation, see [Using Managed Identities locally](#using-managed-identities-locally)).
+
+```python
+# Authentication with Managed Identities
+from azure.identity import DefaultAzureCredential
+
+# Acquire default Azure credentials, which are useful in most circumstances.
+credentials = DefaultAzureCredential()
+```
+
+To use a service principal, acquire a `ServicePrincipalCredential` object. Review [Authenticate with the Azure SDK for Python](/azure/python/python-sdk-azure-authenticate) for information on obtaining a tenant ID, a client ID, and application secret.
```

Comment:
`azure-identity` credential for authenticating a service principal with a secret is `ClientSecretCredential`